core/controllers/reader_test.py

```diff
@@ -135,13 +135,15 @@ def _init_classify_inputs(self, exploration_id):
             exp_services.get_exploration_by_id(exploration_id).states['Home'])
 
     def _is_string_classifier_called(self, answer):
+        sc = classifier_registry.ClassifierRegistry.get_classifier_by_id(
+            feconf.DEFAULT_STRING_CLASSIFIER)
         string_classifier_predict = (
-            lda_string_classifier.LDAStringClassifier.predict)
+            sc.__class__.predict)
```

Comment:
Isn't predict an instance method? I would actually expect the swap on ``sc.predict`` to work here. What's the error if you do it that way?